google/cloud/datastore/query.py

```diff
@@ -688,6 +875,19 @@ def _pb_from_query(query):
     composite_filter = pb.filter.composite_filter
     composite_filter.op = query_pb2.CompositeFilter.Operator.AND
 
+    for filter in query.filters:
+        if isinstance(filter, BaseCompositeFilter):
```

Comment:
this logic looks like it's repeated in a few places - any way we can consolidate?